c2cgeoportal/views/layers.py

```diff
@@ -283,8 +293,27 @@ def security_cb(r, feature, o):
             if allowed.scalar() == 0:
                 raise HTTPForbidden()
 
-        protocol = self._get_protocol_for_layer(layer, before_update=security_cb)
-        return protocol.update(self.request, feature_id)
+            # check is geometry is valid
+            self.validate_geometry_(spatial_elt)
+
+        protocol = self._get_protocol_for_layer(layer, before_update=check_geometry)
+        try:
+            feature = protocol.update(self.request, feature_id)
+            return feature
```

Comment:
Same reason. :) I had to open the source code of `papyrus` to know what `protocol.update()` returns. This variable should make it a bit clearer.